src/components/button/Button.styles.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import Link from 'gatsby-link'
+import styled from 'styled-components'
+import { variant, space, layout, compose } from 'styled-system'
+import { darken, lighten } from 'polished'
+import { colors } from '../../theme/colors'
+import { fonts } from '../../theme/fonts'
+import { useSpaceLayoutProps } from '../../hooks/useSpaceLayoutProps'
+import { ButtonAsAnchorProps, ButtonProps } from './Button.types'
+
+type StyledButtonProps =
+  | (ButtonAsAnchorProps & { renderAs: 'a' | typeof Link })
+  | (ButtonProps & { renderAs: 'button' })
+
+export const StyledButton = styled(
+  ({ renderAs, to, ...props }: StyledButtonProps) => {
+    const spaceLayoutProps = useSpaceLayoutProps(props)
+    if (renderAs === 'a') {
+      // disable eslint rule as children passed as props will automatically render content
+      // eslint-disable-next-line jsx-a11y/anchor-has-content
+      return <a {...spaceLayoutProps[1]} />
+    }
+    if (renderAs === Link) {
+      return <Link to={to!} {...spaceLayoutProps[1]} />
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prop `spaceLayoutProps[1]` must be placed on a new line](https://app.codacy.com/manual/PrideInLondon/pride-london-web/pullRequest?prid=4765408)